### PR TITLE
Release 1.0.2

### PR DESCRIPTION
**Pull request by dylanuys**
### QOL Upgrades

- Auto-update for both validators and miners.
- Self-healing for both validators and miners, restarts neuron every 6 hours to help keep processes in a healthy state
- Reduced validator log clutter 
  - Before: Lots of innocuous warnings and ugly progress bars from pytorch, huggingface, etc. that made logs difficult to parse
  - Now: Silenced warnings and removed progress bars, keeping only information essential to understanding validator behavior/health

### Validator Technical Improvements:

Improved validator image generation pipeline
**Before**
- The validator prompt generation pipline produced annotations with inconsistent white spacing and punctuation between prompt responses.
- Our image generation pipline attempted to generate images from annotations that exceeded the maximum allowed length of 77 tokens for CLIPTokenizer, the tokenizer for the Stable Diffusion XL, Mobius, and RealVisXL_V4.0 diffusion models used by validators. These long annotations previously triggered warnings.
- Previously in the validator forward function, resampling was not performed if a NAN value was present in any field of a real image sample dict (e.g. Image id).

**Now**
- Improved BLIP-2 outputs to be cleaner with spaces between each prompt and periods at the end of responses.
- Synthetic image generation now first truncates all descriptive information into the maximum tokenizer token limit upfront, to ensure compliance with HuggingFace diffusion model tokenizers and avoid warnings.
- Validator forward function now checks for NAN image metadata after each sampling, annotation, and image generation step. A retry with a resampled real image is performed if data is invalid.

### CODE REVIEW

File: bitmind/base/utils/weight_utils.py [hunk 1]
```diff
@@ -157,6 +157,9 @@ def process_weights_for_netuid(
 
     # Find all non zero weights.
     non_zero_weight_idx = np.argwhere(weights > 0).squeeze()
+    if non_zero_weight_idx.ndim == 0:
+        non_zero_weight_idx = non_zero_weight_idx.reshape((1,))
+
```
Comment (dylanuys) [inline bitmind/base/utils/weight_utils.py:162]: this is a fix for an issue i raised a while back for the bittensor-subnet-template repo:
https://github.com/opentensor/bittensor-subnet-template/issues/93

File: bitmind/constants.py [hunk 2]
```diff
@@ -11,8 +11,8 @@
         {"path": "bitmind/celeb-a-hq", "create_splits": False}
     ],
     "fake": [
-        {"path": "bitmind/realvis-xl", "create_splits": False},
-        {"path": "bitmind/stable-diffusion-xl", "create_splits": False},
+        {"path": "bitmind/realvis-xl___parquet", "create_splits": False},
+        {"path": "bitmind/stable-diffusion-xl___parquet", "create_splits": False},
```
Comment (dylanuys) [inline bitmind/constants.py]: we are renaming in huggingface and reverting these to the original names right now

File: install_system_deps.sh [hunk 3]
```diff
@@ -3,5 +3,19 @@ sudo apt install python3-pip -y
...
+WALLET_NAME=default
+WALLET_HOTKEY=default
+MINER_AXON_PORT=8091
+BLACKLIST_FORCE_VALIDATOR_PERMIT=True" > miner.env
+
+echo "NETUID=34
+SUBTENSOR_NETWORK=finney
+WALLET_NAME=default
+WALLET_HOTKEY=default
+VALIDATOR_AXON_PORT=8092" > validator.env
```
Comment (kenobijon) [inline install_system_deps.sh:20]: Do we have any docs about these being the default axon ports?

**Review by kenobijon — APPROVED** [review]
Looks good, have reviewed many of the individual PRs into testnet branch. Small cleanups on documentation but overall good to release

File: install_system_deps.sh (hunk 3 above)
Comment (dylanuys) [reply install_system_deps.sh:20]: @kenobijon  Not explicitly, but our updated MIner/Validator Guide's instruct users to modify the .env files according to their environment.

If you think it'd be helpful to mention explicitly that these are the defaults, i can add that too!

**Pull request merged**